HpToolsLauncher/Runners/AlmTestSetsRunner.cs

```diff
@@ -1411,12 +1411,15 @@ private TestRunResults UpdateTestStatus(TestSuiteRunResults runResults, ITestSet
             ITSTest currentTest = null;
             try
             {
-
                 //find the test for the given status object
                 currentTest = targetTestSet.TSTestFactory[testExecStatusObj.TSTestId];
 
                 //find the test in our list
                 var testIndex = GetIndexOfTestIdentifiedByName(currentTest.Name, runResults);
+                if (testIndex == -1)
+                {
+                    Console.WriteLine("No test index exist for this test");
+                }

```

Comment:
You may have to exit the code block, otherwise the next line `runResults.TestRuns[-1]` will crash the application

Reply:
Thanks for suggestion. Now is fixed.